app/assets/javascripts/angular/controllers/graph_ctrl.js

```diff
@@ -143,7 +147,18 @@ angular.module("Prometheus.controllers").controller('GraphCtrl',
     $scope.addAxis();
   }
 
+  $scope.$watch('graph.range', $scope.refreshGraph);
+  $scope.$watch('graph.endTime', $scope.refreshGraph);
```

Comment:
ah, that is truly spectacular.

the arguments to the callback function are `function(newValue, oldValue)`.
in this case `$scope.refreshGraph` was the function, and `newValue` could be an enormous integer representing the unix time for the endtime. this was interpreted as the debounce time for `$scope.refreshGraph`, hence looooong waits for the refresh to occur. updated.